cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -0,0 +1,515 @@
+   * @return a map of artifact id to plugin classes in the artifact for all plugin classes in the namespace.
+   *         The map will never be null. If there are no plugin classes, an empty map will be returned.
+   * @throws IOException if there was an exception reading metadata from the metastore
+   */
+  public Map<Id.Artifact, List<PluginClass>> getPluginClasses(final Id.Namespace namespace,
+                                                              final String type) throws IOException {
+    return metaTable.executeUnchecked(
+      new TransactionExecutor.Function<DatasetContext<Table>, Map<Id.Artifact, List<PluginClass>>>() {
+        @Override
+        public Map<Id.Artifact, List<PluginClass>> apply(DatasetContext<Table> context) throws Exception {
+          Map<Id.Artifact, List<PluginClass>> result = Maps.newHashMap();
+
+          Scanner scanner = context.get().scan(scanPlugins(namespace, type));
+          Row row;
+          while ((row = scanner.next()) != null) {
+            addPluginsToMap(result, row);
+          }
+          return result;
+        }
+      });
+  }
+
+  /**
+   * Get all plugin classes of a specific type and name in the given namespace.
+   * Results are returned as a map from artifact to plugins in that artifact.
+   *
+   * @param namespace the namespace to look for plugins in
+   * @param type the type of plugin to look for
+   * @param name the name of the plugin to look for
+   * @return a map of artifact id to plugin classes in the artifact for all plugin classes in the namespace.
+   *         The map will never be null. If there are no plugin classes, an empty map will be returned.
+   * @throws IOException if there was an exception reading metadata from the metastore
+   */
+  public Map<Id.Artifact, List<PluginClass>> getPluginClasses(final Id.Namespace namespace, final String type,
+                                                              final String name) throws IOException {
+    return metaTable.executeUnchecked(
+      new TransactionExecutor.Function<DatasetContext<Table>, Map<Id.Artifact, List<PluginClass>>>() {
+        @Override
+        public Map<Id.Artifact, List<PluginClass>> apply(DatasetContext<Table> context) throws Exception {
+          Map<Id.Artifact, List<PluginClass>> result = Maps.newHashMap();
+
+          PluginKey pluginKey = new PluginKey(namespace, type, name);
+          Row row = context.get().get(pluginKey.getRowKey());
+          if (row != null) {
+            addPluginsToMap(result, row);
+          }
+          return result;
+        }
+      });
+  }
+
+  /**
+   * Clear all data in the given namespace. Not terribly efficient, can do more deletes than needed.
+   * This is purely for unit tests, so it's not a big concern.
+   *
+   * @param namespace the namespace to delete data in
+   * @throws IOException if there was some problem deleting the data
+   */
+  @VisibleForTesting
+  void clear(Id.Namespace namespace) throws IOException {
```

Comment:
thats a good point, it needs to be called.  I'll figure out how to expose it later when it is actually in use.